packages/gatsby-plugin-image/src/components/gatsby-image.server.tsx

```diff
@@ -57,6 +58,8 @@ export const GatsbyImage: FunctionComponent<GatsbyImageProps> = function GatsbyI
     }
   }
 
+  // QUESTION-1: sources is not used in getMainProps
+  // I'm not sure to understand the purpose.
```

Comment:
Summoning @ascorbic for this question :)

Reply:
It's assinged to cleanedImages below, which is passed to getMainProps, which then spreads it into the result. This means "sources" is then spread into MainImage

Reply:
Thank you, I've checked the code back and was isolating the test on img, while image is inside the picture tag. Sorry for bothering :/